docs/developer-guide/cstor-volume.md

```diff
@@ -0,0 +1,189 @@
+      X                    X   PVC-YAML      X                |                         |
+    XXXXX                  X                 X                |                         |
+      X        Applies     X                 X                +-------------------------+
+      X     +------------->X                 X
+     XXX                   X                 X
+    X X X                  X                 X
+      X                    X                 X
+                           X                 X
+                           XXXXXXXXXXXXXXXXXXX
+
+```
+2. As part of PVC create, the csi-provisioner(of Controller Plugin pod) executes a gRPC method that is implemented by 
+`cstor-csi-plugin` (of controller plugin pod). This method is `CreateVolume`.
+
+3. The `cstor-csi-plugin` (as part of CreateVolume) creates a CStorVolumeConfig(CVC) custom resource that is reconciled 
+by CVC-Operator. It should be noted that CVC-Operator is deployed as part of installing cStor. 
+
+```
+                                                                    +------------------+
+                                                                    |                  |
++--------------------+                                              |                  |
+| Controller Plugin  |                                              |                  |
+|                    |                                              |                  |
++--------------------+  cstor-csi-plugin  XXXXXXXXX                 |                  |
+|                    |  creates CVC as   X        X                 |   CVC-Operator   |
+|                    |  part of volume  X         X                 |                  |
+| cstor-csi-plugin   |  create request X          X   Reconciles    |                  |
+|                    |                 X          +---------------->+                  |
+|                    +---------------->X   CVC    X                 |                  |
+|                    |                 X          X                 |                  |
+|                    |                 X          X                 |                  |
+|                    |                 X          X                 +------------------+
+|                    |                 XXXXXXXXXXXX
++--------------------+
+
+```
+
+4. Now the CVC-Operator creates following resources as part of CVC reconciliation:
+- A `CStorVolume` (CV) (custom resource).  
+- `k` number of `CStorVolumeReplica` (CVR) (custom resource) where `k` is the replica count specified in the storage class.
+- A deployment known as `cStor target`.
+- A service. 
+
+5. `CStorVolume` replica is reconciled by the corresponding `cStor target` that got created.
+
+6. The created `CStorVolumeReplicas` are placed on `CStorPoolInstances`. Now two `CStorVolumeReplicas` for the given PVC
+will be placed on the same `CStorPoolInstance`(CSPI). (The CVR is labelled with a CSPI UUID -- ths is what placement
+means)
+
```

Comment:
nit: No two `CStorVolumeReplicas` for the given PVC
will be placed on the same `CStorPoolInstance`(CSPI). (The CVR is labelled with a CSPI UUID -- which means replicas will be distributed across the pools)